tests/testthat/test-FilterState.R

```diff
@@ -113,6 +117,36 @@ testthat::test_that("set_state cannot set mutable fields in a fixed FilterState"
   testthat::expect_identical(shiny::isolate(filter_state$get_state()), old_state)
 })
 
+testthat::test_that("set_state can set mutable fields in a locked FilterState", {
+  filter_state <- FilterState$new(c("a", NA_character_), dataname = "data", varname = "variable", locked = TRUE)
+  old_state <- shiny::isolate(filter_state$get_state())
+  new_state <- filter_var(
+    dataname = "data",
+    varname = "variable",
+    selected = "a",
+    keep_na = TRUE,
+    keep_inf = FALSE,
+    disable = FALSE,
+    locked = TRUE
+  )
+  testthat::expect_output(filter_state$set_state(new_state), "WARN.+attempt to disable a locked filter aborted")
```

Comment:
I don't think this should pass as `new_state` is `disabled = FALSE` so no attempt should be made.

Reply:
@chlebowa it changes everything else (updates keep_na, keep_inf, selected) besides `disable`

Reply:
@chlebowa but disabled will be removed with #300